### DISCUSSION TITLE
Java Agent example for debugging applications on Jakarta EE or Java EE servers with ByteBuddy

### DISCUSSION THREAD

**Discussion by mrts** [raphw/byte-buddy#1595] [Show and tell]
Thanks a lot for ByteBuddy! It helped me to debug an issue that only occurred in production in a Java EE application.

However, there were no easy to find examples for the particular oddities of using an agent in a Java EE application server environment (building a fat JAR with dependencies with _maven-assembly-plugin_, not being able to use a logger etc), so I created a small example project here: https://github.com/mrts/java-agent-for-jakarta-ee-server-debugging.

All thoughts and criticism much welcome!

**Comment by raphw** [comment]
Thanks for sharing!
